tools/copyAccessLogFileTool.php

```diff
@@ -175,19 +175,41 @@ function _copyFile($filePath) {
 		}
 
 		// Uncompress it, if needed.
-		$gunzipPath = escapeshellarg(Config::getVar('cli', 'gunzip'));
+		$gunzipPath = Config::getVar('cli', 'gunzip');
 		if ($isCompressed) {
-			exec($gunzipPath . ' ' . $tmpFilePath);
+			if (!is_executable($egrepPath)) {
+				printf(__('admin.copyAccessLogFileTool.error.executingUtil', array('utilPath' => $gunzipPath, 'utilVar' => 'gunzip')) . "\n");
+				exit(1);
+			}
+			$gunzipPath = escapeshellarg($gunzipPath);
+			$output = array();
+			$returnValue = 0;
+			exec($gunzipPath . ' ' . $tmpFilePath, $output, $returnValue);
+			if ($returnValue > 1) {
+				printf(__('admin.copyAccessLogFileTool.error.executingUtil', array('utilPath' => $gunzipPath, 'utilVar' => 'gunzip')) . "\n");
+				exit(1);
+			}
 			$tmpFilePath = substr($tmpFilePath, 0, -3);
 		}
 
 		// Filter only entries that contains journal paths.
-		$egrepPath = escapeshellarg(Config::getVar('cli', 'egrep'));
+		$egrepPath = Config::getVar('cli', 'egrep');
 		$destinationPath = $usageStatsDir . DIRECTORY_SEPARATOR .
 		FILE_LOADER_PATH_STAGING . DIRECTORY_SEPARATOR .
 		pathinfo($tmpFilePath, PATHINFO_BASENAME);
+		if (!is_executable($egrepPath)) {
+			printf(__('admin.copyAccessLogFileTool.error.executingUtil', array('utilPath' => $egrepPath, 'utilVar' => 'egrep')) . "\n");
+			exit(1);
+		}
+		$egrepPath = escapeshellarg(Config::getVar('cli', 'egrep'));
+		$output = array();
+		$returnValue = 0;
 		// Each journal path is already escaped, see the constructor.
-		exec($egrepPath . " -i '" . $this->_journalPaths . "' " . escapeshellarg($tmpFilePath) . " > " . escapeshellarg($destinationPath));
+		exec($egrepPath . " -i '" . $this->_journalPaths . "' " . escapeshellarg($tmpFilePath) . " > " . escapeshellarg($destinationPath), $output, $returnValue);
+		if ($returnValue > 2) {
```

Comment:
egrep manual tells that 2 is returned on error, so we should be checking for equal or greater than 2, I think.